### DISCUSSION TITLE
How to remove keycloak default login form?

### DISCUSSION THREAD

**Discussion by wyhdil** [keycloak/keycloak#23886] [Q&A] [answered]
hello, there are some qusetions about login, can u help me?
 
I have implemented self-hosted gitlab as identify provider in keycloak.
![4d52afd7657d1cf395c8bea92b37b664](https://github.com/keycloak/keycloak/assets/69418800/29d63056-35ad-46ca-8db2-d054478010ea)
At this time, there should be two login methods, one is to log in directly using the keycloak account, and the other is to log in using gitlab authorization.

But now I want to prohibit users from using keycloak accounts to log in directly, and can only use self-hosted gitlab accounts for third-party authorized logins.

So what should I do? I hope you can help me solve it. Thanks！

**Comment by wyhdil** [comment, marked as answer]
Done!

**Reply by retaggi** [reply]
Thank you, how did you do that? :)

**Comment by copper111** [comment]
Nice answer, very useful

**Reply by danielFesenmeyer** [reply]
There is no better answer than "Done" :rofl:
